### PR TITLE
Add 10.7.5 Tray and Installer Changes

### LINKED ISSUE #47: jellyfin tray - open jellyfin point s to local host
Hi all i was looking around as a newbie because I stumbeled over an issue which I cant resolve. When Jellyfin Tray is started the menu gives you the option to open Jellyfin which will be started in a browser window. 
Now updated from 10.6.4 to 10.7.1 starts the browser to localhost. However I changed Jellyfin in networking to an other local ip-adress. What do I do wrong or is there a way to point this open Jellyfin to another changed IP address?  Thanks in advance for help or suggestions.

### PR DESCRIPTION
**Pull request by anthonylavado**
Bringing this repo up to date with changes from the shipped version of 10.7.5.

Enhancements:
- Uses local bind address in launching browser (fixes #47)
- Adds shortcut to desktop properly
- Updates .vscode launch and build tasks with correct SDK name
- Adds some more text warnings to installer about Service Mode and Data Dir location (more to follow)
- Bumps version

Build failures in CI are to be expected, those CI scripts need a bunch of updating that hasn't been done yet.

### CODE REVIEW

File: Jellyfin.Windows.Tray/TrayApplicationContext.cs [hunk 1]
```diff
@@ -122,7 +125,16 @@ private void LoadPortFromJellyfinConfig()
...
-                    _localJellyfinUrl = "http://localhost:" + port + "/web/index.html";
+                    XDocument networkXml = XDocument.Load(_networkFile);
+                    XPathNavigator NetworkReader = networkXml.CreateNavigator();
+
+                    NetworkAddress = NetworkReader.SelectSingleNode("/NetworkConfiguration/LocalNetworkAddresses").Value;
+
+                    if (string.IsNullOrEmpty(NetworkAddress)) {
+                        _localJellyfinUrl = "http://localhost:" + port + "/web/index.html";
+                    } else {
+                    _localJellyfinUrl = "http://" + NetworkAddress + ":"+  port + "/web/index.html";
```
Comment (nielsvanvelzen) [inline Jellyfin.Windows.Tray/TrayApplicationContext.cs:136]:
```suggestion
                        _localJellyfinUrl = "http://" + NetworkAddress + ":"+  port + "/web/index.html";
```

File: Jellyfin.Windows.Tray/TrayApplicationContext.cs [hunk 2]
```diff
@@ -18,7 +18,9 @@ public class TrayApplicationContext : ApplicationContext
         private readonly string _jellyfinServiceName = "JellyfinServer";
         private readonly string _autostartKey = "JellyfinTray";
         private string _configFile;
+        private string _networkFile;
         private bool FirstRunDone = false;
+        private string NetworkAddress;
```
Comment (nielsvanvelzen) [inline Jellyfin.Windows.Tray/TrayApplicationContext.cs:23]:
```suggestion
        private string _networkAddress;
```

Comment (nielsvanvelzen) [reply Jellyfin.Windows.Tray/TrayApplicationContext.cs:23]: Also it looks like it is only used locally so no need for a class property I guess

File: Jellyfin.Windows.Tray/TrayApplicationContext.cs [hunk 3]
```diff
@@ -122,7 +125,16 @@ private void LoadPortFromJellyfinConfig()
...
                     string port = SettingsReader.SelectSingleNode("/ServerConfiguration/PublicPort").Value;
 
-                    _localJellyfinUrl = "http://localhost:" + port + "/web/index.html";
+                    XDocument networkXml = XDocument.Load(_networkFile);
+                    XPathNavigator NetworkReader = networkXml.CreateNavigator();
+
+                    NetworkAddress = NetworkReader.SelectSingleNode("/NetworkConfiguration/LocalNetworkAddresses").Value;
+
+                    if (string.IsNullOrEmpty(NetworkAddress)) {
+                        _localJellyfinUrl = "http://localhost:" + port + "/web/index.html";
```
Comment (nielsvanvelzen) [inline Jellyfin.Windows.Tray/TrayApplicationContext.cs:134]: If you, instead, set the `NetworkAddress` to `localhost` here you can prevent duplicating this string concatenation.

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: Jellyfin.Windows.Tray/TrayApplicationContext.cs (hunk 2 above)
Comment (anthonylavado) [reply Jellyfin.Windows.Tray/TrayApplicationContext.cs]: Applied change, will fix the rest in other commit.

File: Jellyfin.Windows.Tray/.vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (dkanada) [inline Jellyfin.Windows.Tray/.vscode/launch.json:4]:
```suggestion
```

File: Jellyfin.Windows.Tray/.vscode/tasks.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                "watch",
+                "run",
+                "${workspaceFolder}/Jellyfin.Windows.Tray.csproj",
+                "/property:GenerateFullPaths=true",
+                "/consoleloggerparameters:NoSummary"
+            ],
+            "problemMatcher": "$msCompile"
+        }
+    ]
+}
```
Comment (dkanada) [inline Jellyfin.Windows.Tray/.vscode/tasks.json:42]:
```suggestion
}

```

File: Jellyfin.Windows.Tray/TrayApplicationContext.cs [hunk 6]
```diff
@@ -106,24 +109,41 @@ private void CreateTrayIcon()
...
                     XDocument systemXml = XDocument.Load(_configFile);
                     XPathNavigator SettingsReader = systemXml.CreateNavigator();
 
                     FirstRunDone = SettingsReader.SelectSingleNode("/ServerConfiguration/IsStartupWizardCompleted").ValueAsBoolean;
-                    string port = SettingsReader.SelectSingleNode("/ServerConfiguration/PublicPort").Value;
+                    var publicPort = SettingsReader.SelectSingleNode("/ServerConfiguration/PublicPort")?.Value;
+                    if (!string.IsNullOrEmpty(publicPort))
+                    {
+                        _port = publicPort;
+                    }
```
Comment (crobibero) [inline Jellyfin.Windows.Tray/TrayApplicationContext.cs:132]: Should also attempt to pull this from network.xml also?

Comment (anthonylavado) [reply Jellyfin.Windows.Tray/TrayApplicationContext.cs:132]: Yeah, it should. This PR is big enough, I will add it in the next bit.

**Pull request merged**